packages/react-bindings/test/styles/resolveStylesAndClasses-test.ts

```diff
@@ -52,17 +54,83 @@ describe('resolveStylesAndClasses', () => {
     const renderStyles = jest.fn().mockReturnValue('a')
     const { classes } = resolveStylesAndClasses(componentStyles, styleParam, renderStyles)
 
-    expect(classes.root).toBeDefined()
+    expect(classes['__root']).toBeDefined()
     expect(renderStyles).toHaveBeenCalledWith({ color: 'red' })
   })
 
   test('caches rendered classes', () => {
     const renderStyles = jest.fn().mockReturnValue('a')
     const { classes } = resolveStylesAndClasses(componentStyles, styleParam, renderStyles)
 
-    expect(classes.root).toBeDefined()
+    expect(classes['__root']).toBeDefined()
+    expect(renderStyles).toHaveBeenCalledWith({ color: 'red' })
+    expect(classes['__root']).toBeDefined()
+    expect(renderStyles).toHaveBeenCalledTimes(1)
+  })
+
+  test('caches resolved styles for no props', () => {
+    spyOn(componentStyles, 'root').and.callThrough()
+    const { resolvedStyles } = resolveStylesAndClasses(componentStyles, styleParam, () => '', true, 'Test', theme, {})
+    const { resolvedStyles: secondResolvedStyles } = resolveStylesAndClasses(componentStyles, styleParam, () => '', true, 'Test', theme, {})
+
+    expect(resolvedStyles.root).toMatchObject({ color: 'red' })
+    expect(componentStyles.root).toHaveBeenCalledTimes(1)
+    expect(secondResolvedStyles.root).toMatchObject({ color: 'red' })
+    expect(componentStyles.root).toHaveBeenCalledTimes(1)
+  })
+
+  test('caches classes for no props', () => {
+    const renderStyles = jest.fn().mockReturnValue('a')
+    const { classes } = resolveStylesAndClasses(componentStyles, styleParam, renderStyles, true, 'Test1', theme, {})
```

Comment:
I have to change the display name here in order to avoid colliding of the cache with previous examples. Other options is to provide different theme (maybe this is better), or to add some option for invalidating the cache...